mesh/profiles/views.py

```diff
@@ -19,28 +19,39 @@ def bio_view(request):
         return JsonResponse({'error': request.method + ' Method not allowed'}, status=405)
 
 
-def profile_picture(request):
-    if request.method == "GET":
-        data = request.GET
-        response = {
-            "status": "success"
-        }
-        if data.get("accountID") is None:
-            response.update({"status": "error"})
-            response.update({"message": "Missing account ID."})
-            return JsonResponse(response)
-        else:
-            try:
-                profile = Profile.objects.get(accountID=int(data.get("accountID")))
-                response.update({
-                    "data": {
-                        "get": {
-                            "profilePicture": profile.image.url
-                        }
-                    }
-                })
-                return JsonResponse(response)
-            except ObjectDoesNotExist:
-                response.update({"status": "error"})
-                response.update({"message": "An account does not exist with this account ID."})
-                return JsonResponse(response)
+class ProfilePicturesView(View):
+    def get(self, request, account_id, *args, **kwargs):
+        return get_data(account_id, "profilePicture", lambda profile: profile.profilePicture.url)
+
+
+class UserNamesView(View):
+    def get(self, request, account_id, *args, **kwargs):
+        return get_data(account_id, "userName", lambda profile: profile.userName)
+
+
+class PreferredNamesView(View):
+    def get(self, request, account_id, *args, **kwargs):
+        return get_data(account_id, "preferredName", lambda profile: profile.preferredName)
+
+
+class PreferredPronounsView(View):
+    def get(self, request, account_id, *args, **kwargs):
+        return get_data(account_id, "preferredPronouns", lambda profile: profile.preferredPronouns)
+
+
+def get_data(account_id, name, mapper):
+    try:
+        profile = Profile.objects.get(accountID=int(account_id))
```

Comment:
Here is where you'd call the validate json, and youd pass in the request body im pretty sure